### PR TITLE
llvmdev 17.0.1

### LINKED ISSUE #227: Set LLVM_ENABLE_DUMP=ON with LLVM 17+
This enables rusticl, but might change the ABI. So it's better to wait until a new version of LLVM comes out.

### PR DESCRIPTION
**Pull request by h-vetinari**
Just a simple bump from what's already in the `rc`-channel.

Closes #227, see https://github.com/conda-forge/llvmdev-feedstock/pull/238/commits/da3c0ebe6168c2ffb93062daf9a4ef58db0444da
Closes #239

### CODE REVIEW

File: recipe/bld.bat [hunk 1]
```diff
@@ -12,7 +12,9 @@ cmake -G "Ninja" ^
     -DCMAKE_BUILD_TYPE="Release" ^
     -DCMAKE_PREFIX_PATH=%LIBRARY_PREFIX% ^
     -DCMAKE_INSTALL_PREFIX:PATH=%LIBRARY_PREFIX% ^
+    -DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreadedDLL ^
```
Comment (h-vetinari) [inline recipe/bld.bat:15]: See https://github.com/conda-forge/llvmdev-feedstock/pull/238/commits/d7a15d807ffd85c5033054d443cb8d53c7f649a0

File: recipe/bld.bat [hunk 2]
```diff
@@ -23,6 +25,7 @@ cmake -G "Ninja" ^
     -DLLVM_INCLUDE_TESTS=ON ^
     -DLLVM_INCLUDE_UTILS=ON ^
     -DLLVM_INSTALL_UTILS=ON ^
+    -DLLVM_USE_SYMLINKS=OFF ^
```
Comment (h-vetinari) [inline recipe/bld.bat:28]: See https://github.com/conda-forge/llvmdev-feedstock/pull/238/commits/bea09a1c4770bc496190b3ac228121c4c5c3ef75

**Review by h-vetinari — COMMENTED** [review]
Some context for the changes.

**Comment by h-vetinari** [comment]
OK, this took a few final fits and starts, but is now ready for review @conda-forge/llvmdev 

PTAL 🙏

**Comment by h-vetinari** [comment]
Gentle ping @conda-forge/llvmdev

**Comment by isuruf** [comment]
Thanks

**Pull request merged**